database/database.go

```diff
@@ -77,3 +77,46 @@ func InsertUserProfile(user *model.UserProfile) error {
 	}
 	return err
 }
+
+//GenerateToken func to add token in db
+func GenerateToken(username string) error {
+	var userAuth *model.UserAuth
+	var hashedtoken string
+	var err error
+	hashedtoken, err = EncryptPassword(username)
+	if err == nil {
+		userAuth.Token = hashedtoken
+	} else {
+		log.Print(err)
+		return err
+	}
+	userAuth.Username = username
+	stmt, err := db.Prepare("INSERT INTO user_auth(Username,Token) VALUES(?,?)")

```

Comment:
Use user_profile_id (id of user profile table) instead of username in the table.
Because in case a user changes his/her username, you will only need to change it in one place (user_profile) table.